src/stream.ts

```diff
@@ -107,203 +99,123 @@ export class YamuxStream implements Stream {
     this.epochStart = Date.now()
     this.getRTT = init.getRTT
 
-    this.abortController = new AbortController()
-
     this.sendFrame = init.sendFrame
-    this.onStreamEnd = init.onStreamEnd
-
-    this.sourceInput = pushable({
-      onEnd: (err?: Error) => {
-        if (err != null) {
-          this.log?.error('stream source ended id=%s', this._id, err)
-        } else {
-          this.log?.trace('stream source ended id=%s', this._id)
-        }
 
-        this.closeRead()
-      }
+    this.source = each(this.source, () => {
+      this.sendWindowUpdate()
     })
```

Comment:
How does this handle errors?

Previously we had:
```ts
 try {
      for await (const val of this.sourceInput) {
        this.sendWindowUpdate()
        yield val
      }
    } catch (err) {
      const errCode = (err as { code: string }).code
      if (errCode !== ERR_STREAM_ABORT) {
        this.log?.error('stream source error id=%s', this._id, err)
        throw err
      }
    }
```

Reply:
Previously the source would not throw if the error was `ERR_STREAM_ABORT` - this type of error is now not thrown.

If the stream is aborted an error must be passed to `.abort(err)`, and that error is the one that is thrown, so the try/catch is redundant.